src/partition/repart.c

```diff
@@ -4995,6 +5005,174 @@ static int context_open_copy_block_paths(
         return 0;
 }
 
+static int fd_apparent_size(int fd, uint64_t *ret) {
```

Comment:
there should be a comment here explaining why this is not using fstat()'s st_blocks fields.